google/cloud/spanner_v1/merge_result_set.py

```diff
@@ -0,0 +1,114 @@
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass
+from queue import Queue
+from typing import Any, TYPE_CHECKING
+from threading import Lock
+
+if TYPE_CHECKING:
+    from google.cloud.spanner_v1.database import BatchSnapshot
+
+QUEUE_SIZE_PER_WORKER = 32
+LOCK = Lock()
+
+
+def _set_metadata(merged_result_set, results):
+    LOCK.acquire()
+    try:
+        merged_result_set._metadata = results.metadata
+        merged_result_set._stats = results.stats
```

Comment:
Actually, when I think of it: We should either not add stats at all, or otherwise add the stats from all calls. The stats could contain things like number of rows returned. That value is only relevant if we sum all the values from all the stats that are returned for all the partitions.

My suggestion would be to maybe delay this for a separate PR.

Reply:
stats is only used for getting the number of rows updated. IMO it would always be 0 for partitioned query as its a select type of query so should we return -1 for stats always here?

Reply:
Actually, stats can also be returned for queries if you set the query mode to `PROFILE`. The stats will then contain the number of rows, but also stats about the query execution time.

Reply:
Cursor only used stats to get row count. It does not return other details of stats. 
Added a TODO to implement stats and returning None for stats in this PR

Reply:
Ack